[Song Mengfei] ip

Song-Mengfei opened the pull request:
# Duke Pro!
>“Your mind is for having ideas, not holding them.” – _David Allen_ ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
* easy to learn
+ ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/Song-Mengfei/ip).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

**And it is FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you have any question, feel free to leave comment. 😀

If you want to use the application, please run the `main` method in `Launcher.java`

```Java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Song-Mengfei committed:
Rename, Greet, Exit

Song-Mengfei committed:
Echo

Song-Mengfei committed:
Add, List

Song-Mengfei committed:
Add, List(Fixed)

Song-Mengfei committed:
Mark as Done

Song-Mengfei committed:
ToDos, Events, Deadlines

Song-Mengfei committed:
Automated Text UI Testing

Song-Mengfei committed:
Handle Errors

Song-Mengfei committed:
Delete

Song-Mengfei committed:
With Javadoc

Song-Mengfei committed:
Add factory method

Song-Mengfei committed:
Save

Song-Mengfei committed:
Dates and Times

Song-Mengfei committed:
More OOP and Packages

Song-Mengfei committed:
Merge branch 'add-gradle-support'

Song-Mengfei committed:
Gradle and JUnit

Song-Mengfei committed:
JavaDoc

Song-Mengfei committed:
Follow the Coding Standard

Song-Mengfei committed:
Find

Song-Mengfei committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/command/AddDeadline.java
#	src/main/java/duke/command/AddEvent.java
#	src/main/java/duke/command/AddTodo.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/Delete.java
#	src/main/java/duke/command/Unknown.java
#	src/main/java/duke/exception/DukeException.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/tasklist/Deadline.java
#	src/main/java/duke/tasklist/Event.java
#	src/main/java/duke/tasklist/Task.java
#	src/main/java/duke/tasklist/TaskList.java
#	src/main/java/duke/tasklist/Todo.java
#	src/main/java/duke/ui/UI.java

Song-Mengfei committed:
Recover files

Song-Mengfei committed:
Little fix

Song-Mengfei committed:
no message

Song-Mengfei committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/UI/UI.java
#	src/main/java/duke/command/AddDeadline.java
#	src/main/java/duke/command/AddEvent.java
#	src/main/java/duke/command/CommonMethods.java
#	src/main/java/duke/command/Find.java
#	src/main/java/duke/exception/DukeException.java
#	src/main/java/duke/taskList/Task.java
#	src/main/java/duke/taskList/TaskList.java

Song-Mengfei committed:
change name

Song-Mengfei committed:
Debug

Song-Mengfei committed:
Debug

Song-Mengfei committed:
Merge branch 'master' of https://github.com/Song-Mengfei/ip

pra-navi commented:
i could not see the italicised part, also not sure if i should be able to edit the checklist but right now I am unable to do so, but everything else looks great!

Song-Mengfei committed:
GUI

YeoBohShin commented on data/duke.txt:
```diff
@@ -0,0 +1,3 @@
+0 | return book | 2001-10-29
+1 |  read book
+0 |  return book
```
Shouldn't duke.txt be added to gitignore.?

YeoBohShin commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,52 @@
+ * A utility class for parsing user input and creating corresponding command objects.
+ * This class processes user input to determine the type of command to create.
+ */
+public class Parser {
```
i like the way to organised the code here, easy to read.

YeoBohShin commented on src/main/java/duke/command/AddEvent.java:
```diff
@@ -0,0 +1,63 @@
+     * @throws DukeException If there is an error while executing the command.
+     */
+    @Override
+    public String execute(TaskList lst, UI io, Storage storage) throws DukeException {
+        if (s.isEmpty() || s.equals(" ")) {
+            throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+        } else if (!s.matches(" \\S.*\\s/from\\s\\d.*\\s/to\\s\\d.*")) {
```
Should use a more descriptive variable name instead of s here.

YeoBohShin reviewed (commented):
Overall not much error with your code. Good work!

pra-navi commented on src/main/java/duke/command/AddDeadline.java:
```diff
@@ -0,0 +1,60 @@
+     *
+     * @param s The input string containing the task description and deadline.
+     */
+    public AddDeadline(String s) {
```
minor but perhaps can consider using the parameter name as 'string' for more clarity? that way later when u use it in methods such as execute it will be more clear

pra-navi commented on src/main/java/duke/command/Remark.java:
```diff
@@ -0,0 +1,58 @@
+                storage.changeFile(lst);
+                return io.unmark(t);
+            }
+        } catch (IOException ignored) {
```
i suppose it might be more intuitive to call the exception 'exception' instead of ignored?

pra-navi commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,52 @@
+    public static Command parse(String instr) {
+        String keyWord = instr.split(" ")[0];
+        switch (keyWord) {
+        case"":
```
i like the use of switch-case and the correct coding standard applied according to this course!

pra-navi commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,124 @@
+            FileWriter fw = new FileWriter(filePath, true);
+            fw.write(t.getText() + "\n");
+            fw.close();
+        } catch (IOException iE) {
```
perhaps it might be a good idea to uniformly name your exceptions? for clarity

pra-navi reviewed (commented):
LGTM. Just a few nits to fix. Great documentation and organisation of the code!

Song-Mengfei committed:
branch10

Song-Mengfei committed:
fix: change build.gradle

The elder version cannot run on MacOS

Let's
*add the 'javafx.fxml' package in build.gradle
*add the checkstyle function in build.gradle

It is the easiest way for every one who get this repo could
install the package

Song-Mengfei committed:
Some class are not following the coding rule

Improve the code qulity|

It is easier for others to read and understande

Song-Mengfei committed:
Merge pull request #1 from Song-Mengfei/branch-A-CodeQuality

Improve Code Quality

Song-Mengfei committed:
Use Assertions

Song-Mengfei committed:
Merge pull request #2 from Song-Mengfei/branch-A-Assertions

Use Assertions

Song-Mengfei committed:
Add snooze function

In previous versions, tasks can not be snoozed or paused
and the chatbox won't close after recieving "bye"

Add a Snooze command
Fix the other functions
Add instructions to close the chatbox
Change the path of duke

Song-Mengfei committed:
Merge pull request #3 from Song-Mengfei/B-Snooze

Add snooze function

Song-Mengfei committed:
A unique personality

Song-Mengfei committed:
fix the error that the window won't close after "bye"

Song-Mengfei committed:
Adding Ui.png

Song-Mengfei committed:
User Guide

Song-Mengfei committed:
update .gitignore

Song-Mengfei committed:
update ./gitignore

Song-Mengfei committed:
More error handling

Song-Mengfei committed:
more checkstyle

Song-Mengfei committed:
More error handling

Song-Mengfei committed:
More and more error handling

Song-Mengfei committed:
Make a directory when there is no "/data" directory

Song-Mengfei committed:
solve the hard-coded file path

Song-Mengfei committed:
Solving following issue

Song-Mengfei committed:
debug

Song-Mengfei committed:
Better GUI

Song-Mengfei committed:
More more debug

Song-Mengfei committed:
Fix the errors on Linux
